### PR TITLE
Fix tests for Aug 2023 updated remote datasets

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

Some tests related to the GMT remote datasets are failing due to the recent updates of these datasets (see https://www.generic-mapping-tools.org/remote-datasets/changes.html for a list of these changes). The min/max values of these datasets change slightly in the new versions.

This PR fixes these failing tests by updating the min/max values.

**Notes:**

1. In the tests, we use statements like `npt.assert_allclose(data.min(), -180.40002, rtol=1e-5)` to compare the grid value with the expected value. All datasets have a specific precision (e.g., the `earth_geoid` dataset has a precision of 0.01), thus it makes no sense to compare `earth_geoid` value with a number like `-180.40002` and a relative tolerance of 1.0e-5.  Instead, it makes more sense to compare the value with `-180.4` and an absolute tolerance of 0.01 (the precision of the earth_geoid dataset). I've changed `rtol` to `atol` in all these tests.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by seisman** [comment]
Two synbath tests are not updated due to upstream dataset issue (https://github.com/GenericMappingTools/gmtserver-admin/issues/213).

**Comment by weiji14** [comment]
> Two synbath tests are not updated due to upstream dataset issue ([GenericMappingTools/gmtserver-admin#213](https://github.com/GenericMappingTools/gmtserver-admin/issues/213)).

Ok, looks they're re-processing the synbath dataset again, let's wait a day or two to see if it gets fixed.

File: pygmt/tests/test_datasets_earth_relief.py [hunk 1]
```diff
@@ -37,8 +37,8 @@ def test_earth_relief_01d_igpp_synbath(data_source):
     assert data.shape == (181, 361)
     npt.assert_allclose(data.lat, np.arange(-90, 91, 1))
     npt.assert_allclose(data.lon, np.arange(-180, 181, 1))
-    npt.assert_allclose(data.min(), -8600.5)
-    npt.assert_allclose(data.max(), 5559.0)
+    npt.assert_allclose(data.min(), -8600.5, atol=0.5)
+    npt.assert_allclose(data.max(), 5559.0, atol=0.5)
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:41]: Did you get the uncertainty for the earth_relief grids from the paper(s)? I'm not seeing them on https://www.generic-mapping-tools.org/remote-datasets/earth-relief.html#technical-information.

File: pygmt/tests/test_datasets_earth_relief.py [hunk 2]
```diff
@@ -52,8 +52,8 @@ def test_earth_relief_01d_gebco(data_source):
     assert data.gmt.registration == 0
     npt.assert_allclose(data.lat, np.arange(-90, 91, 1))
     npt.assert_allclose(data.lon, np.arange(-180, 181, 1))
-    npt.assert_allclose(data.min(), -8598)
-    npt.assert_allclose(data.max(), 5559.0)
+    npt.assert_allclose(data.min(), -8597.0, atol=1.0)
+    npt.assert_allclose(data.max(), 5559.0, atol=1.0)
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:56]: Same here for the GEBCO grids, is the 1.0 uncertainty value from some paper? Not seeing it at https://www.generic-mapping-tools.org/remote-datasets/earth-gebco.html.

File: pygmt/tests/test_datasets_earth_relief.py [hunk 3]
```diff
@@ -52,8 +52,8 @@ def test_earth_relief_01d_gebco(data_source):
     assert data.gmt.registration == 0
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:52]: Metadata checks for gebco/gebcosi seems to have been left out from #1818
```suggestion
    assert data.gmt.registration == 0
    assert data.attrs["units"] == "meters"
    assert data.attrs["long_name"] == "Earth elevation relative to the geoid"
    assert data.attrs["vertical_datum"] == "EGM96"
    assert data.attrs["horizontal_datum"] == "WGS84"
```

**Review by weiji14 — COMMENTED** [review]
> 1. In the tests, we use statements like `npt.assert_allclose(data.min(), -180.40002, rtol=1e-5)` to compare the grid value with the expected value. All datasets have a specific precision (e.g., the `earth_geoid` dataset has a precision of 0.01), thus it makes no sense to compare `earth_geoid` value with a number like `-180.40002` and a relative tolerance of 1.0e-5.  Instead, it makes more sense to compare the value with `-180.4` and an absolute tolerance of 0.01 (the precision of the earth_geoid dataset). I've changed `rtol` to `atol` in all these tests.

Cool, setting the absolute tolerance does makes more sense. Looked through each dataset and the precision values look ok, except for the ones indicated below where I couldn't find the precision documented at https://www.generic-mapping-tools.org/remote-datasets/index.html.

I'm wondering if we should file an upstream issue to https://github.com/GenericMappingTools/remote-datasets/issues to have the precision information in the metadata too? That way we could do something like `npt.assert_allclose(data.min(), ..., atol=data.attrs["precision"])` instead of hardcoding a value. Just in case the precision changes in the future.

**Comment by seisman** [comment]
> Looked through each dataset and the precision values look ok, except for the ones indicated below where I couldn't find the precision documented at https://www.generic-mapping-tools.org/remote-datasets/index.html.

I get the precisions from the recipes（https://github.com/GenericMappingTools/gmtserver-admin/tree/master/recipes）that were used in building the grids (the `DST_SCALE` parameter in the recipes).

> I'm wondering if we should file an upstream issue to [https://github.com/GenericMappingTools/remote-datasets/issues](https://github.com/GenericMappingTools/remote-datasets/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) to have the precision information in the metadata too? That way we could do something like `npt.assert_allclose(data.min(), ..., atol=data.attrs["precision"])` instead of hardcoding a value. Just in case the precision changes in the future.

I'm not sure, because these values are just the precisions that GMT used to build the grids, not exactly the real precisions of the original datasets.

**Comment by weiji14** [comment]
> > Looked through each dataset and the precision values look ok, except for the ones indicated below where I couldn't find the precision documented at https://www.generic-mapping-tools.org/remote-datasets/index.html.
> 
> I get the precisions from the recipes（https://github.com/GenericMappingTools/gmtserver-admin/tree/master/recipes) that were used in building the grids (the `DST_SCALE` parameter in the recipes).
> 
> > I'm wondering if we should file an upstream issue to [https://github.com/GenericMappingTools/remote-datasets/issues](https://github.com/GenericMappingTools/remote-datasets/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) to have the precision information in the metadata too? That way we could do something like `npt.assert_allclose(data.min(), ..., atol=data.attrs["precision"])` instead of hardcoding a value. Just in case the precision changes in the future.
> 
> I'm not sure, because these values are just the precisions that GMT used to build the grids, not exactly the real precisions of the original datasets.

Ah I see, if it's just GMT's re-gridded precisions and not the actual precisions, then it probably shouldn't be in the metadata. Let's just wait for the grids to be fixed then, seems like the `earth_synbath_*` grids have been fixed on the candidate server already, and just need to wait for them to be copied to the oceania server according to https://github.com/GenericMappingTools/gmtserver-admin/issues/213#issuecomment-1694670286.

File: pygmt/tests/test_datasets_earth_relief.py [hunk 4]
```diff
@@ -117,8 +117,8 @@ def test_earth_gebcosi_15m_with_region():
     assert data.gmt.registration == 1
     npt.assert_allclose(data.lat, np.arange(-87.875, -84, 0.25))
     npt.assert_allclose(data.lon, np.arange(85.125, 87, 0.25))
-    npt.assert_allclose(data.min(), -531)
-    npt.assert_allclose(data.max(), 474)
+    npt.assert_allclose(data.min(), -531, atol=1.0)
+    npt.assert_allclose(data.max(), 474, atol=1.0)
 
 
 def test_earth_relief_30s_synbath():
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:124]: Synbath grid seems to be updated now. Change L135:136 to:

```
    npt.assert_allclose(data.min(), -3552.5, atol=0.5)
    npt.assert_allclose(data.max(), -2257.5, atol=0.5)
```

File: pygmt/tests/test_datasets_earth_relief.py [hunk 5]
```diff
@@ -69,8 +69,8 @@ def test_earth_relief_01d_with_region_srtm():
     assert data.gmt.registration == 0
     npt.assert_allclose(data.lat, np.arange(-5, 6, 1))
     npt.assert_allclose(data.lon, np.arange(-10, 11, 1))
-    npt.assert_allclose(data.min(), -5154)
-    npt.assert_allclose(data.max(), 805.5)
+    npt.assert_allclose(data.min(), -5154, atol=0.5)
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:72]:
```suggestion
    npt.assert_allclose(data.min(), -5151, atol=0.5)
```

File: pygmt/tests/test_datasets_earth_relief.py [hunk 6]
```diff
@@ -244,8 +244,8 @@ def test_earth_relief_15s_default_registration():
     npt.assert_allclose(data.coords["lat"].data.max(), 4.997917)
     npt.assert_allclose(data.coords["lon"].data.min(), -9.997917)
     npt.assert_allclose(data.coords["lon"].data.max(), -9.502083)
-    npt.assert_allclose(data.min(), -3897)
-    npt.assert_allclose(data.max(), -74)
+    npt.assert_allclose(data.min(), -3897, atol=0.5)
+    npt.assert_allclose(data.max(), -74, atol=0.5)
 
 
 def test_earth_relief_03s_default_registration():
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:251]: Change L263:264 to:

```
    npt.assert_allclose(data.min(), -2070.0, atol=0.5)
    npt.assert_allclose(data.max(), -924.5, atol=0.5)
```

File: pygmt/tests/test_datasets_earth_relief.py [hunk 7]
```diff
@@ -244,8 +244,8 @@ def test_earth_relief_15s_default_registration():
     npt.assert_allclose(data.coords["lat"].data.max(), 4.997917)
     npt.assert_allclose(data.coords["lon"].data.min(), -9.997917)
     npt.assert_allclose(data.coords["lon"].data.max(), -9.502083)
-    npt.assert_allclose(data.min(), -3897)
-    npt.assert_allclose(data.max(), -74)
+    npt.assert_allclose(data.min(), -3897, atol=0.5)
+    npt.assert_allclose(data.max(), -74, atol=0.5)
```
Comment (weiji14) [inline pygmt/tests/test_datasets_earth_relief.py:248]:
```suggestion
    npt.assert_allclose(data.max(), -71, atol=0.5)
```

**Review by weiji14 — COMMENTED** [review]
May need to manually regenerate the cache by uncommenting https://github.com/GenericMappingTools/pygmt/blob/dcf1c7f5d4ecb98361a0e03473a91f3c1dcec7e5/.github/workflows/cache_data.yaml#L15 before re-running the tests.

**Comment by weiji14** [comment]
If you're short on time, I can help push the changes? Just give me a :+1:

File: pygmt/tests/test_datasets_earth_relief.py (hunk 3 above)
Comment (weiji14) [reply pygmt/tests/test_datasets_earth_relief.py:56]: Done at 74fa646130fb1586d828abc8197b26ab4a6a0213.

**Review by weiji14 — APPROVED** [review]
Have updated the cache at 364c1b5b6f21f07e985a2f231a4c20f1d7ad059d and re-ran the tests with updated minmax numbers, should all work now.

**Pull request merged**